lib/azure/armrest/armrest_manager.rb

```diff
@@ -155,10 +155,10 @@ def subscription_info(subscription_id = @subscription_id)
 
       # Returns a list of resource groups for the given subscription.
       #
-      def resource_groups
+      def resource_groups(subscription_id = @subscription_id)
         url = url_with_api_version(@base_url, 'subscriptions', subscription_id, 'resourcegroups')
         resp = rest_get(url)
```

Comment:
Is that possible? I wouldn't think so.

I did think about wrapping http errors and re-raising them as some sort of armrest error, but that's down the road a bit.